src/decision-engine/index.js

```diff
@@ -112,7 +123,7 @@ class DecisionEngine {
 
     // If there's nothing in the message, bail out
     if (msg.empty) {
-      this._requestQueue.tasksDone(peerId, tasks)
+      peerId && this._requestQueue.tasksDone(peerId, tasks)
```

Comment:
`peerId` is obtained on line 76 via following code:

```js
 const { peerId, tasks, pendingSize } = this._requestQueue.popTasks(this._opts.targetMessageSize)
```

However `popTasks` returns an object without `peerId` in 2 code paths out of 3. At the same time `tasksDone` and `messageSent` below do not expect `undefined`.

It was not exactly clear what made most sense here, so I just conditioned those calls on `peerId`s presence, but it would be good to make sure all this makes sense.